### PR TITLE
Added temporary workaround for storing `ioa_rule_groups` in `Policy`

### PR DESCRIPTION
**Pull request by swedgwood**
# Temporary workaround for `ioa_rule_groups` in `Policy`

- [ ] Enhancement
- [ ] Major feature update
- [x] Bug fixes
- [ ] Breaking change
- [ ] Updated unit tests
- [ ] Documentation

## Bandit analysis

```shell
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.10.5
Run started:2022-07-20 15:33:51.235798

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 0
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0
                Low: 0
                Medium: 0
                High: 0
        Total issues (by confidence):
                Undefined: 0
                Low: 0
                Medium: 0
                High: 0
```

## Issues resolved

- `Policy.dump` claims that it conforms to the CrowdStrike API Spec, but it in the case where `style="prevention"`, it does not conform as doesn't as it doesn't have an `ioa_rule_groups` key. In the future I assume we'll have a proper wrapper for `ioa_rule_groups`, but as a temporary work around to keep `dump`'s claims true this should work. I've also added TODOs so hopefully this won't be forgotten when it comes to implementing that wrapper.

## Other

- I encountered this whilst working on tests in #30

### CODE REVIEW

File: caracara/common/policy_wrapper.py [hunk 1]
```diff
@@ -281,6 +283,10 @@ class Policy:
     settings_groups: List[PolicySettingGroup] = None
     settings_key_name = "settings"
 
+    # TODO: Replace with a proper wrapper around ioa_rule_groups, this is temporary so that this
+    #       result is still reachable from the 'dump' method when style="prevention".
+    _raw_ioa_rule_groups: dict = None
```
Comment (ChristopherHammond13) [inline caracara/common/policy_wrapper.py:288]: What data types does the dictionary map? Can we do something like this?
```_raw_ioa_rule_groups: Dict[str, str] = None```

**Review by ChristopherHammond13 — CHANGES_REQUESTED** [review]
Just one proposed change and this will be good to merge. Thank you, @swedgwood!!

File: caracara/common/policy_wrapper.py (hunk 1 above)
Comment (swedgwood) [reply caracara/common/policy_wrapper.py:288]: Now that I'm looking at the documentation (ctrl+f for `responses.PreventionPolicyV1` [here](https://assets.falcon.crowdstrike.com/support/api/swagger.html)), I meant to put something like `List[dict]`. Even still though, `List[Dict[str, str]]` wouldn't be right because, looking at the docs (`responses.IOARuleGroupV1`), the value type could be `str`, `bool`, or `int`. I guess I could do `List[Dict[str, Union[str, bool, int]]]`, but if I'm going that far I might as well use a `TypedDict` and specify the whole schema. Maybe we could do `List[Dict[str, Any]]` as a good middle ground?

Comment (ChristopherHammond13) [reply caracara/common/policy_wrapper.py:288]: ```List[Dict[str, Any]]``` is a perfect middle ground :)
